### PR TITLE
Add ksonnet template for http proxy

### PR DESCRIPTION
**Pull request by yupbank**
with the http-proxy image, now user can finally test the inception model without client requirements

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/228)

### CODE REVIEW

**Comment by jlewi** [comment]
@elsonrodriguez Do you want to review this since you worked on the inception example?

**Comment by yupbank** [comment]
That's a good point !

**Comment by yupbank** [comment]
image

**Comment by jlewi** [comment]
The incorrect report of test failure is 
kubeflow/testing#30

File: kubeflow/tf-serving/tf-serving.libsonnet [hunk 1]
```diff
@@ -90,6 +94,26 @@ local networkSpec = networkPolicy.mixin.spec;
                     // model-server doesn't have something we can use out of the box.
                     resources: defaults.resources,
                   },
+		  if httpProxyImage != 0 then
+		  {
+		    name: name+"-http-proxy",
+                    image: httpProxyImage,
+                    imagePullPolicy: defaults.imagePullPolicy,
+                    command: [
+                      "/bin/sh",
```
Comment (jlewi) [inline kubeflow/tf-serving/tf-serving.libsonnet]: Why do we we have to wrap the command in a shell? Why can't we just do

command: [
  "python",
  "/usr/src/app/server.py",
   ...
]

**Comment by jlewi** [comment]
@yupbank Test flakiness should be fixed if you sync and rerun the tests.

**Comment by yupbank** [comment]
`ERROR|2018-02-20T20:30:42|/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py|192| Time out waiting for Workflows kubeflow-presubmit-legacy-228-f25ebb0-275-7615 to finish
`

**Comment by jlewi** [comment]
The test failed:
http://testing-argo.kubeflow.io/timeline/kubeflow-test-infra/kubeflow-presubmit-legacy-228-f25ebb0-276-4e92;tab=workflow

+ git clone https://github.com/kubeflow/kubeflow.git /mnt/test-data-volume/kubeflow-presubmit-legacy-228-f25ebb0-276-4e92/src/kubeflow/kubeflow
fatal: destination path '/mnt/test-data-volume/kubeflow-presubmit-legacy-228-f25ebb0-276-4e92/src/kubeflow/kubeflow' already exists and is not an empty dire
ctory.

Don't know why the directory would already exist.

**Comment by jlewi** [comment]
I confirmed the directory `/mnt/test-data-volume/kubeflow-presubmit-legacy-228-f25ebb0-276-4e92/src/kubeflow/kubeflow` does exist and does contain kubeflow/kubeflow code.

On Retry the test appears to have succeeded but TFJob test actually failed. 

This is kubeflow/testing#27

**Comment by jlewi** [comment]
Test failure is #273 so unrelated to this PR.

**Comment by jlewi** [comment]
Nice addition; thank you!

**Pull request merged**